irc/client.go

```diff
@@ -214,6 +215,8 @@ type Session struct {
 	batch MultilineBatch
 
 	webPushEndpoint string // goroutine-local: web push endpoint registered by the current session
+
+	metadataSubscriptions []string
```

Comment:
Actually how do you feel about making this `utils.HashSet[string]`? Sorry for the churn